manifest.go

```diff
@@ -59,6 +59,31 @@ func (f ManifestFile) parse() (manifest, error) {
 }
 
 func (m manifest) path(s string) (string, error) {
+	r, err := m.pathExact(s)
+	if err == nil || err == ErrEmpty {
+		return r, err
+	}
+	// If path references a runfile that lies under a directory that itself is a
+	// runfile, then only the directory is listed in the manifest. Look up all
+	// prefixes of path in the manifest.
+	prefixEnd := len(s)
+	for {
+		prefixEnd = strings.LastIndex(s[0:prefixEnd-1], "/")
```

Comment:
Maybe path.Split helps here to make this a tiny bit shorter.

Reply:
It does. I still have to trim the trailing slash, but the code seems simpler now.